content/basics/use-ably.textile

```diff
@@ -0,0 +1,77 @@
+
+The **realtime interface** is most commonly used client-side and is stateful. It establishes a connection to Ably and maintains state for the life of the connection. Reasons to use the realtime interface include:
+
+* You are developing a mobile, desktop or web client that needs to subscribe to messages in real time.
+* You want to maintain a persistent connection to Ably, attach to one or more channels, and publish and subscribe to messages.
+* Your application needs to register its presence on a channel, or listen for others becoming present in real time.
+* You want to publish at very high message rates or with the lowest possible realtime latencies.
+
+The **REST interface** is most commonly used server-side and is stateless. Reasons to use the REST interface include:
+
+* Your application server is used to primarily issue tokens for clients or publish messages on channels.
+* Your application is mostly stateless, for example you process a request or respond to an event, and then move onto the next request or event without any previous state carrying through.
+* You prefer a synchronous request over an asynchronous request. Note not all REST libraries are synchronous, but where the platform offers a synchronous and asynchronous approach, the REST libraries are more often synchronous.
+* You just want to publish data to the platform and don't need to subscribe or manage channel attachment.
+* You want to get one-time info about the presence set of a channel.
+
+h2(#rest-api). Raw REST API
+
+Ably provides a raw "REST API":/api/rest-api that doesn't require an SDK to use. Ably recommends using the REST interface where possible but you may want to use the raw REST API if your chosen platform isn't supported or you have resourcing constraints.
+
+h2(#protocol-adapters). Protocol adapters
+
+Ably's "protocol adapters":/protocols are an alternative method for connection to Ably. Whilst Ably SDKs are the recommended method for connecting to Ably, protocol adapters provide interoperability with a wide variety of client libraries. 
+
+The following protocol adapters are available:
+
+* "External MQTT libraries":#mqtt
+* "Ably SSE and EventSource API":#sse-eventsource
+* "Pusher and PubNub protocols":#pusher-pubnub
+
+h3(#mqtt). External MQTT libraries
+
+The Ably "MQTT protocol adapter":/protocols/mqtt translates between "MQ Telemetry Transport (MQTT)":/https://mqtt.org and Ably’s own protocol. MQTT is a publish/subscribe, lightweight messaging protocol designed for constrained devices and low-bandwidth networks.
+
+Ably recommends you use MQTT when:
+
+* Using devices which do not have a device-specific Ably SDK, such as Arduino platforms or in C/C++ applications
+* If bandwidth is limited and you want to keep network traffic to a minimum
```

Comment:
```suggestion
h3(#mqtt). MQTT

The Ably "MQTT protocol adapter":/protocols/mqtt translates between the "Message Queuing Telemetry Transport (MQTT)":/https://mqtt.org and Ably’s own protocol. MQTT is a lightweight messaging protocol optimized for high-latency and unreliable networks. It is often used in remote devices with small footprints. MQTT libraries are available for most internet-enabled devices.

You can use the MQTT protocol adapter with your MQTT library when:

* an Ably SDK isn't available for your target platform
* bandwidth is limited and you want to keep network traffic to a minimum
* you want to avoid any type of vendor lock-in by using open protocols
```